projects/ngx-prx-styleguide/src/lib/tz-datepicker/tz-datepicker.component.ts

```diff
@@ -14,21 +14,27 @@ const moment = momentNs;
       <prx-datepicker
         name="date"
         [date]="this.model.pickerDate"
-        (dateChange)="handleDateChange($event)"
+        (dateChange)="this.model.pickerDate = $event; handleChange()"
         [changed]="changed"
       >
       </prx-datepicker>
       <ng-container *ngIf="!supportsTimeInput">
         <input
           [ngClass]="{ changed: changed }"
           (change)="handleChange()"
```

Comment:
that's a lotta change handling. Sorry if my earlier suggestions weren't clear. To clean this up and use the ngForms, go back to double binding the `([ngModel])` but emit with the `(ngModelChange)="handleChange()"` event. Don't use both `change` and `ngModelChange`. If using FormsModule/ngModel, prefer ngModelChange over DOM events.

Same thing for the select below.

Template driven forms aren't quite as dated as Todd Motto saying they are made of "awesome sauce" but he does have a pretty basic article on them: https://ultimatecourses.com/blog/angular-2-forms-template-driven
Reactive forms on the other hand, give you an observable for changes and such and are the newer, much better standard.